### PR TITLE
dlb: support virtual DLB devices

### PR DESCRIPTION
**Pull request by daixiang0**
Upgrade DLB dirver version to support virtual DLB devices.

Risk Level: Low

### CODE REVIEW

**Comment by YiXR** [comment]
This patch fixes the problem that dlb siov and sriov cannot work properly on envoy, and helps dlb run on alibaba ecs server. Hope the code can be integrated as soon as possible.

**Review by phlax — APPROVED** [review]
mobile fails are unrelated (#27339 )

lgtm, from deps pov, thanks @daixiang0 

would be good to get a stamp from a c++ maintainer before landing

**Comment by ggreenway** [comment]
@daixiang0 CI had some issues. Can you merge main and see if that resolves it?

**Comment by daixiang0** [comment]
@ggreenway thanks, I convert to draft before CI happy.

**Comment by daixiang0** [comment]
@phlax I do not know how to fix https://source.cloud.google.com/results/invocations/5d918406-b722-4b14-8e18-419a5a3fdcc1/targets/%2F%2Fdocs:html/log, could you give some clues?

File: bazel/repository_locations.bzl
Comment (phlax) [inline bazel/repository_locations.bzl:1355]:
```suggestion
        urls = ["https://downloadmirror.intel.com/763709/dlb_linux_src_release8.0.0.txz"],
```

this @daixiang0

Comment (daixiang0) [reply bazel/repository_locations.bzl:1355]: Cool!

**Comment by daixiang0** [comment]
@ggreenway please take a look :)

**Review by ggreenway — APPROVED** [review]
LGTM

**Comment by daixiang0** [comment]
ping @phlax

**Review by phlax — APPROVED** [review]
lgtm, thanks @daixiang0

**Comment by daixiang0** [comment]
thank you very much @phlax

**Pull request merged**
